File: scrape_bardahl.js
const fs = require('fs');

async function scrapeBardahl() {
    const urls = [
        "https://bardahl.ma/collections/moto",
        "https://bardahl.ma/collections/moto?page=2"
    ];
    let products = [];

    for (const url of urls) {
        try {
            const response = await fetch(url);
            const html = await response.text();

            // Simple regex to extract products since we can't use DOMParser in vanilla Node without JSDOM
            // We look for card-wrapper links and images
            const cardRegex = /<li class="grid__item"[^>]*>([\s\S]*?)<\/li>/g;
            let match;

            while ((match = cardRegex.exec(html)) !== null) {
                const cardHtml = match[1];

                // Extract title
                const titleMatch = cardHtml.match(/<a[^>]*class="full-unstyled-link"[^>]*>([\s\S]*?)<\/a>/);
                const title = titleMatch ? titleMatch[1].trim() : null;

                // Extract image
                let imgUrl = null;
                const imgMatch = cardHtml.match(/<img[^>]*srcset="([^"]*)"/);
                if (imgMatch) {
                    const srcset = imgMatch[1];
                    const firstImg = srcset.split(',')[0].trim().split(' ')[0];
                    imgUrl = "https:" + firstImg.split('?')[0] + "?width=823";
                } else {
                    const srcMatch = cardHtml.match(/<img[^>]*src="([^"]*)"/);
                    if (srcMatch) {
                        imgUrl = "https:" + srcMatch[1].split('?')[0] + "?width=823";
                    }
                }

                if (title && imgUrl && !products.find(p => p.title === title)) {
                    products.push({ title, image: imgUrl });
                }
            }
        } catch (e) {
            console.error("Error fetching", url, e);
        }
    }

    fs.writeFileSync('bardahl_products.json', JSON.stringify(products, null, 2));
    console.log(`Scraped ${products.length} products.`);
}

scrapeBardahl();

File: scrape_bardahl.py
import urllib.request
import re
from bs4 import BeautifulSoup
import json

urls = [
    "https://bardahl.ma/collections/moto",
    "https://bardahl.ma/collections/moto?page=2"
]

products = []

for url in urls:
    req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
    response = urllib.request.urlopen(req)
    html = response.read()
    soup = BeautifulSoup(html, 'html.parser')
    
    # Based on Shopify default dawn theme structure
    product_cards = soup.select('.card-wrapper')
    
    for card in product_cards:
        try:
            title_el = card.select_one('.full-unstyled-link')
            if not title_el:
                continue
            title = title_el.text.strip()
            
            img_el = card.select_one('img')
            img_url = ""
            if img_el and img_el.has_attr('srcset'):
                # Extract the first image from srcset or src
                img_url = "https:" + img_el['src'].split('?')[0] + "?width=823"
            elif img_el and img_el.has_attr('src'):
                img_url = "https:" + img_el['src'].split('?')[0] + "?width=823"
                
            if title and img_url:
                products.append({
                    "title": title,
                    "image": img_url
                })
        except Exception as e:
            pass

print(json.dumps(products, indent=2))
